src/datapump/jobs/version_update.py

```diff
@@ -0,0 +1,186 @@
+from enum import Enum
+from typing import List, Optional, Union, Dict
+
+from pydantic import StrictInt
+
+from ..clients.data_api import DataApiClient
+from ..jobs.jobs import Job, JobStatus
+from ..util.exceptions import DataApiResponseError
+
+
+class NonNumericFloat(str, Enum):
+    nan = "nan"
+
+
+NoDataType = Union[StrictInt, NonNumericFloat]
+
+
+class VersionUpdateJobStep(str, Enum):
+    starting = "starting"
+    creating_tile_set = "creating_tile_set"
+    creating_tile_cache = "creating_tile_cache"
+    creating_aux_assets = "creating_aux_assets"
+    mark_latest = "mark_latest"
+
+
+class VersionUpdateJob(Job):
```

Comment:
I would use a name more like "RasterTileCacheUpdateJob", since this is very specific to update raster tile caches, but in the future we might have more jobs that create new versions on the data api using different pipelines (e.g. I'm going to add a burned areas job that will upload a vector table and create a vector tile cache).